python/google/protobuf/internal/message_test.py

```diff
@@ -2441,5 +2442,36 @@ def testSucceedOversizeProto(self):
     q.ParseFromString(self.p_serialized)
     self.assertEqual(self.p.field.payload, q.field.payload)
 
+
+@unittest.skipIf(api_implementation.Type() != 'cpp' or
+                 sys.version_info < (2, 7),
+                 'explicit tests of the C++ implementation for PY27 and above')
+class DeepNestedProtosTest(BaseTestCase):
+
+  def setUp(self):
+    # create a Tree whose depth exceeds the default allowed recursive depth
+    # when reading
+    root = recursive_test_pb2.Node(value=1)
+    for x in range(199):
+        root = recursive_test_pb2.Node(tree=recursive_test_pb2.Tree(right=root))
+
+    self.p = recursive_test_pb2.Tree(right=root)
+    self.p_serialized = self.p.SerializeToString()
+
+  def testAssertOversizeProto(self):
+    from google.protobuf.pyext._message import SetAllowOversizeProtos
+    SetAllowOversizeProtos(False)
+    q = recursive_test_pb2.Tree
```

Comment:
The test won't pass without parentheses:
```suggestion
    q = recursive_test_pb2.Tree()
```